src/aws-cpp-sdk-core/include/smithy/tracing/RootTraceProbe.h

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#pragma once
+
+#include <smithy/Smithy_EXPORTS.h>
+#include <smithy/tracing/TraceEvent.h>
+#include <smithy/tracing/TraceProbe.h>
+#include <aws/core/utils/memory/stl/AWSVector.h>
+
+namespace smithy {
+    namespace components {
+        namespace tracing {
+            /**
+             * A specific Implementation of TraceProbe that keeps track
+             * of a list of TraceProbes, that will dispatch a list of events
+             * to each probe. The intention is that this class will be used
+             * as the root entry point to emitting traces to a list of
+             * arbitrary backends.
+             */
+            class SMITHY_API RootTraceProbe : public TraceProbe {
+            public:
+                void dispatchEvents(const Aws::Vector<TraceEvent> &events) override;
+
```

Comment:
whats the ownership model? does it make sense for this to be unique pointer thats moved to the class

Reply:
good call, making it unique ptr so that the root probe will own it